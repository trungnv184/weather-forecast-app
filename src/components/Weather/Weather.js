import React from "react";
import Card from "react-bootstrap/Card";
import PropTypes from "prop-types";

const Weather = ({day, min, max}) => {
  return (
    <Card bg="light">
      <Card.Header data-testid="day">{day}</Card.Header>
      <Card.Body>
        <Card.Text data-testid="min-temp">Min: {min}</Card.Text>
        <Card.Text data-testid="max-temp">Max: {max}</Card.Text>
      </Card.Body>
    </Card>
  );
};

Weather.propTypes = {
  day: PropTypes.string,
  min: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  max: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
};

export default Weather;
