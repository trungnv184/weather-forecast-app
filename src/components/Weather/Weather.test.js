import React from "react";
import {render, cleanup} from "@testing-library/react";
import {screen} from "@testing-library/dom";
import Weather from "./Weather";

afterEach(cleanup);

test("should display all data of weather when input all props", () => {
  render(<Weather day="Monday" min={25} max={30} />);
  expect(screen.getByTestId("day")).toHaveTextContent("Monday");
  expect(screen.getByTestId("min-temp")).toHaveTextContent("Min: 25");
  expect(screen.getByTestId("max-temp")).toHaveTextContent("Max: 30");
});

test("should display empty day when forgot input day prop", () => {
  render(<Weather min={25} max={30} />);
  expect(screen.getByTestId("day")).toBeEmpty();
  expect(screen.getByTestId("min-temp")).toHaveTextContent("Min: 25");
  expect(screen.getByTestId("max-temp")).toHaveTextContent("Max: 30");
});

test("snapshot", () => {
  const {container} = render(<Weather />);
  expect(container.firstChild).toMatchSnapshot();
});
