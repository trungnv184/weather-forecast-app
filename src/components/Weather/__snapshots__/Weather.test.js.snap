// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  class="card bg-light"
>
  <div
    class="card-header"
    data-testid="day"
  />
  <div
    class="card-body"
  >
    <p
      class="card-text"
      data-testid="min-temp"
    >
      Min: 
    </p>
    <p
      class="card-text"
      data-testid="max-temp"
    >
      Max: 
    </p>
  </div>
</div>
`;
